C/services/common/include/builtin_rule.h

```diff
@@ -76,6 +79,62 @@ class BuiltinRule
 {
 	public:
 		typedef enum { StateCleared, StateTriggered } TRIGGER_STATE;
+		class TriggerInfo
+		{
+
+			public:
+				TriggerInfo()
+				{
+					// Store seconds and microseconds
+					gettimeofday(&m_timestamp, NULL);
+					// Create datetime UTC string with microseconds
+					setUTCDateTimeMicro();
+				};
+
+				TRIGGER_STATE
+					getState() const { return m_state; };
+				const std::string&
+					getAssets() const { return m_assets; };
+				const std::string&
+					getUTCDateTime() const { return m_dateTimeUTC; };
+				void	getUTCTimestamp(struct timeval *tm) { *tm = m_timestamp; };
+
+			private:
+				void setUTCDateTimeMicro()
+				{
+					// Populate tm structure with UTC time
+					struct tm timeinfo;
+					gmtime_r(&m_timestamp.tv_sec, &timeinfo);
```

Comment:
This is not meant to be the current time, but rather the timestamp of the item that triggered the notification. These could be different due to buffering effects.

Reply:
Added timestamp_$asset in plugin_eval JSON
This way plugin_reason can fetch that timestamp.

Timestamp set is last reading timestamp in the data buffer